### PR TITLE
Sync `transformers` and `accelerate` versions

### PR DESCRIPTION
**Pull request by michaelbenayoun**
# What does this PR do?

This PR synchronizes `optimum-neuron` with more recent `transformers` and `accelerate` versions:

- `accelerate==0.29.2`, which is the latest release when this PR is being done,
- `transformers==4.40.2`, which will be the latest releae when this PR is merged.

Related PR in `transformers`: https://github.com/huggingface/transformers/pull/30259

On top of that:

- The workflows for Trainium instances have been updated and use K8 now.

### CODE REVIEW

File: .github/workflows/test_trainium_common.yml [hunk 1]
```diff
@@ -16,28 +16,48 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
   cancel-in-progress: true
 
-
 jobs:
   optimum-neuron-tests:
-    runs-on: [self-hosted, 1-aws-trn, 8-cpu, ci] # run the job on the newly created runner
+    name: Run common tests on Trainium 1
+    # runs-on: [self-hosted, 1-aws-trn, 8-cpu, ci] # run the job on the newly created runner
```
Comment (dacorvo) [inline .github/workflows/test_trainium_common.yml:22]:
```suggestion
```

File: optimum/neuron/trainers.py [hunk 2]
```diff
@@ -382,34 +422,43 @@ def prediction_step(
...
-            if self.args.mp_plugin.should_parallelize:
+            if model_parallel_is_initialized():
                 dp_size = get_data_parallel_size()
+            else:
+                dp_size = xm.xrt_world_size()
 
-                tr_loss_div = tr_loss / dp_size
+            tr_loss_div = tr_loss / dp_size
+
+            xm.mark_step()
```
Comment (dacorvo) [inline optimum/neuron/trainers.py:442]: I love when these lines pop magically to solve a sync issue ... 😜

**Review by dacorvo — COMMENTED** [review]
Awesome ! Looks good to me, but found a few nits !

**Comment by dacorvo** [comment]
You should unpin the safetensor package here because there is now a conflict:
https://github.com/huggingface/optimum-neuron/blob/1e7d0f5ae47fd51b2418b1355a2e819e58b69890/text-generation-inference/server/pyproject.toml#L17

File: optimum/neuron/trainers.py [hunk 3]
```diff
@@ -1324,13 +1374,14 @@ def train(
...
         **kwargs,
     ):
-        with hub_neuronx_cache("training", entry=self.model_cache_entry):
-            result = super().train(
-                resume_from_checkpoint=resume_from_checkpoint,
-                trial=trial,
-                ignore_keys_for_eval=ignore_keys_for_eval,
-                **kwargs,
-            )
+        # with hub_neuronx_cache("training", entry=self.model_cache_entry):
```
[the remark this answers is not in the document]
Comment (michaelbenayoun) [reply optimum/neuron/trainers.py:1377]: Fixed, it was a quick test artifact

File: tests/test_trainers.py [hunk 4]
```diff
@@ -12,420 +12,400 @@
...
-        num_eval_samples = 100
-        per_device_eval_batch_size = 16
-        dummy_eval_dataset = create_dummy_dataset({"x": (1,), "labels": (1,)}, num_eval_samples)
+# LLAMA_V2_MODEL_NAME = "michaelbenayoun/llama-2-tiny-16layers-32kv-heads-random"
+# MODEL_NAME = "michaelbenayoun/llama-2-tiny-4kv-heads-4layers-random"
+MODEL_NAME = "michaelbenayoun/llama-2-tiny-4kv-heads-16layers-random"
+MODEL_NAME = "michaelbenayoun/llama-2-tiny-4kv-heads-8layers-random"
+MODEL_NAME = "michaelbenayoun/llama-2-tiny-4kv-heads-4layers-random"
+# MODEL_NAME = "michaelbenayoun/llama-2-tiny-4kv-heads-2layers-random"
+# MODEL_NAME = "hf-internal-testing/tiny-random-GPTNeoForCausalLM"
```
[the remark this answers is not in the document]
Comment (michaelbenayoun) [reply tests/test_trainers.py]: Done!

**Review by dacorvo — APPROVED** [review]
LGTM, thanks ! If you can fix the seq2seq tracing issue before merging that pull-request that is even better.

**Comment by michaelbenayoun** [comment]
I fixed all but one test.

**Pull request merged**